### PR TITLE
snowkuma Planck keymap

### PR DESCRIPTION
**Pull request by snowkuma**
Wide Colemak layout uses minimalist set of keys.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/snowkuma/config.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#ifndef CONFIG_USER_H
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/snowkuma/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`?

It's simpler, and less prone to user error, so if you could make this switch?

```suggestion
#pragma once
```

File: keyboards/planck/keymaps/snowkuma/config.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#define CONFIG_USER_H
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/snowkuma/config.h]:
```suggestion
```

File: keyboards/planck/keymaps/snowkuma/config.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+#include "config_common.h"
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/snowkuma/config.h]: `keyboards/planck/config.h` includes `config_common.h` already, so this line is redundant.

```suggestion
```

File: keyboards/planck/keymaps/snowkuma/config.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+// Settings for homerow mods
+#define TAPPING_TERM 250
+#define IGNORE_MOD_TAP_INTERRUPT
+
+
+// Add the leader key feature
+#define LEADER_TIMEOUT 300
+
+#endif
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/snowkuma/config.h]:
```suggestion
```

File: keyboards/planck/keymaps/snowkuma/custom_keycodes.h [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+#define M_REST LCA(KC_BSPC)
+
+// Shortcuts
+#define INPUT_L LCAG(KC_SPC)
+#define TXT_PLS LGUI(KC_PLUS)
+#define TXT_MIN LGUI(KC_MINS)
+#define SC_CAPF LGUI(LSFT(KC_3))  // Capture the full screen to file
+#define SC_CAPP LGUI(LSFT(KC_4))  // Capture portion of screen to file
+ 
+// Convenience Keycodes
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/snowkuma/custom_keycodes.h]: QMK defines aliases for these keycodes automatically. If you use the already-defined aliases, Lines 36-38 can be deleted.

https://github.com/qmk/qmk_firmware/blob/a120a000d481ed14a87988383be46c5a76604bb6/quantum/quantum_keycodes.h#L29-L31

File: keyboards/planck/keymaps/snowkuma/keymap.c [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+ *
+ * https://github.com/snowkuma
+ *
+ * Colemak is the default layer and I'll add some notes for other changes as I 
+ * go along.
+ * 
+ * version 0.1
+*/
+
+#include "planck.h"
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/snowkuma/keymap.c]: This should be:

```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/planck/keymaps/snowkuma/keymap.c [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+#include "action_layer.h"
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/snowkuma/keymap.c]: ... which includes `action_layer.h` automatically.

```suggestion
```

File: keyboards/planck/keymaps/snowkuma/config.h (hunk 1 above)
Comment (snowkuma) [reply keyboards/planck/keymaps/snowkuma/config.h]: Many thanks for the suggestions I have amended the files as recommended.

File: keyboards/planck/keymaps/snowkuma/keymap.c [hunk 8]
```diff
@@ -0,0 +1,380 @@
...
+  _REGEX,
+  _NUMBER,
+  _ARRANGE,
+  _FUNCTION,
+  _MOUSE,
+  _THUMB
+};
+
+enum planck_keycodes {
+  COLEMAK = SAFE_RANGE,
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/snowkuma/keymap.c]: As you aren't using the `COLEMAK` keycode, you could actually safely delete this definition, and its logic (Lines 301-306).

```suggestion
```

File: keyboards/planck/keymaps/snowkuma/keymap.c [hunk 9]
```diff
@@ -0,0 +1,380 @@
...
+  EMAIL,
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/snowkuma/keymap.c]:
```suggestion
  EMAIL = SAFE_RANGE,
```

**Comment by noroadsleft** [comment]
Also, we recommend against making commits to your master branch when working on QMK, because of the potential for merge conflicts. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to use your master and development branches here:

[**Best Practices: Your fork's master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_best_practices?id=your-fork39s-master-update-often-commit-never)

If you need any help with this just ask. **You do not need to close this pull request.** Please continue to make any commits related to this pull request to your master branch until your pull request is merged.

File: keyboards/planck/keymaps/snowkuma/keymap.c (hunk 8 above)
Comment (snowkuma) [reply keyboards/planck/keymaps/snowkuma/keymap.c]: That makes sense.  It is unused as I'm not switching between alpha layouts at all.  Many thanks.

**Comment by snowkuma** [comment]
Thanks for all your help and recommendations, I really appreciate them.  I will definitely use development branches in any future updates.

**Pull request merged**